libraries/ESP8266WiFiMesh/src/TcpIpMeshBackend.h

```diff
@@ -0,0 +1,283 @@
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+// ESP-NOW is faster for small data payloads (up to a few kB, split over multiple messages). Transfer of up to 234 bytes takes 4 ms.
+// In general ESP-NOW transfer time can be approximated with the following function: transferTime = ceil(bytesToTransfer / 234.0)*3 ms.
+// If you only transfer 234 bytes at a time, this adds up to around 56kB/s. Finally a chance to relive the glory of the olden days
+// when people were restricted to V90 dial-up modems for internet access!
+// TCP-IP takes longer to connect (around 1000 ms), and an AP has to disconnect all connected stations in order to transfer data to another AP, 
+// but this backend has a much higher data transfer speed than ESP-NOW once connected (100x faster or so).
+
+#ifndef __TCPIPMESHBACKEND_H__
+#define __TCPIPMESHBACKEND_H__
+
+#include <WiFiClient.h> 
+#include <WiFiServer.h>
+#include <functional>
+#include <vector>
+#include "MeshBackendBase.h"
+#include "TcpIpNetworkInfo.h"
+
+class TcpIpMeshBackend : public MeshBackendBase {
+
+public:
+
+  /**
+   * TCP/IP constructor method. Creates a TCP/IP node, ready to be initialised.
+   *
+   * @param requestHandler The callback handler for dealing with received requests. Takes a string as an argument which
+   *          is the request string received from another node and returns the string to send back.
+   * @param responseHandler The callback handler for dealing with received responses. Takes a string as an argument which
+   *          is the response string received from another node. Returns a transmission status code as a transmission_status_t.
+   * @param networkFilter The callback handler for deciding which WiFi networks to connect to.
+   * @param meshPassword The WiFi password for the mesh network.
+   * @param ssidPrefix The prefix (first part) of the node SSID.
+   * @param ssidSuffix The suffix (last part) of the node SSID.
+   * @param verboseMode Determines if we should print the events occurring in the library to Serial. Off by default. This setting is separate for each TcpIpMeshBackend instance.
+   * @param meshWiFiChannel The WiFi channel used by the mesh network. Valid values are integers from 1 to 13. Defaults to 1.
+   *                    WARNING: The ESP8266 has only one WiFi channel, and the the station/client mode is always prioritized for channel selection.
+   *                    This can cause problems if several mesh instances exist on the same ESP8266 and use different WiFi channels. 
+   *                    In such a case, whenever the station of one mesh instance connects to an AP, it will silently force the 
```

Comment:
It is believed that the dropout of the connected stations happens due to the channel change announcement being bugged in the SDK (one extra announcement), see [this reference](https://github.com/espressif/ESP8266_NONOS_SDK/issues/94).